### PR TITLE
EZP-31680: Added UDW Search language filter

### PR DESCRIPTION
**Pull request by barw4**
Jira ticket: https://jira.ez.no/browse/EZP-31680

image

Related `ezpublish-kernel` PR: https://github.com/ezsystems/ezpublish-kernel/pull/3043

### CODE REVIEW

File: src/modules/universal-discovery/components/search/search.component.js [hunk 1]
```diff
@@ -37,6 +38,7 @@ export default class SearchComponent extends Component {
         }
 
         const searchText = this._refSearchInput.value;
+        const language = this._refSearchLanguage.value;
```
Comment (dew326) [inline src/modules/universal-discovery/components/search/search.component.js]:
```suggestion
        const languageCode = this._refSearchLanguage.value;
```

File: src/modules/universal-discovery/components/search/search.component.js [hunk 2]
```diff
@@ -49,7 +51,7 @@ export default class SearchComponent extends Component {
             () => ({ isSearching: true, lastSearchText: searchText }),
             () => {
                 const promise = new Promise((resolve) =>
-                    this.props.findContentBySearchQuery(this.props.restInfo, query, resolve, this.props.searchResultsLimit)
+                    this.props.findContentBySearchQuery(this.props.restInfo, query, resolve, this.props.searchResultsLimit, language)
```
Comment (dew326) [inline src/modules/universal-discovery/components/search/search.component.js]:
```suggestion
                    this.props.findContentBySearchQuery(this.props.restInfo, query, resolve, this.props.searchResultsLimit, languageCode)
```

File: src/modules/universal-discovery/components/search/search.component.js [hunk 3]
```diff
@@ -86,6 +88,33 @@ export default class SearchComponent extends Component {
...
+     * Renders the language filter
+     *
+     * @method renderLanguageFilter
+     */
+    renderLanguageFilter() {
+        const selectAttrs = {
+            className: 'form-control',
+            ref: this.setSearchLanguageRef
+        };
+
```
Comment (dew326) [inline src/modules/universal-discovery/components/search/search.component.js]:
```suggestion
```

File: src/modules/universal-discovery/components/search/search.component.js [hunk 4]
```diff
@@ -86,6 +88,33 @@ export default class SearchComponent extends Component {
...
+        const { languages } = this.props;
```
Comment (dew326) [inline src/modules/universal-discovery/components/search/search.component.js]: Please move this to the top of the method. Destructs of `state` or `props` should be on the top.

File: src/modules/universal-discovery/components/search/search.component.js [hunk 5]
```diff
@@ -86,6 +88,33 @@ export default class SearchComponent extends Component {
...
+
```
Comment (dew326) [inline src/modules/universal-discovery/components/search/search.component.js]:
```suggestion
```

File: src/modules/universal-discovery/components/search/search.component.js [hunk 6]
```diff
@@ -86,6 +88,33 @@ export default class SearchComponent extends Component {
...
+        const options = languages.priority.map((value) => {
+            const language = languages.mappings[value];
+            return <option value={language.languageCode}>{language.name}</option>;
```
Comment (dew326) [inline src/modules/universal-discovery/components/search/search.component.js:105]:
```suggestion

            return <option value={language.languageCode}>{language.name}</option>;
```

File: src/modules/universal-discovery/components/search/search.component.js [hunk 7]
```diff
@@ -86,6 +88,33 @@ export default class SearchComponent extends Component {
...
+        });
+
+        return (
+            <div className="c-search-language__select-wrapper">
```
Comment (dew326) [inline src/modules/universal-discovery/components/search/search.component.js]: This is not proper BEM usage
```suggestion
            <div className="c-search__language-wrapper">
```

File: src/modules/universal-discovery/components/search/search.component.js [hunk 8]
```diff
@@ -200,7 +239,7 @@ export default class SearchComponent extends Component {
             { query: lastSearchText },
             'search'
         );
-
+
```
Comment (dew326) [inline src/modules/universal-discovery/components/search/search.component.js:241]:
```suggestion

```

File: src/modules/universal-discovery/components/search/search.component.js (hunk 4 above)
Comment (dew326) [reply src/modules/universal-discovery/components/search/search.component.js]: You should also add `languages` to the `propTypes`

File: src/modules/universal-discovery/services/universal.discovery.service.js [hunk 9]
```diff
@@ -196,11 +196,13 @@ export const findLocationsByParentLocationId = (
...
  * @param {Function} callback
  * @param {Number} limit
+ * @param {String} LanguageCode
  */
-export const findContentBySearchQuery = ({ token, siteaccess }, query, callback, limit = QUERY_LIMIT) => {
+export const findContentBySearchQuery = ({ token, siteaccess }, query, callback, limit = QUERY_LIMIT, languageCode = null) => {
     const body = JSON.stringify({
         ViewInput: {
             identifier: `udw-locations-by-search-query-${query.FullTextCriterion}`,
+            languageCode: languageCode,
```
Comment (dew326) [inline src/modules/universal-discovery/services/universal.discovery.service.js]:
```suggestion
            languageCode,
```

File: src/modules/universal-discovery/universal.discovery.module.js [hunk 10]
```diff
@@ -813,7 +813,11 @@ export default class UniversalDiscoveryModule extends Component {
         } = this.props;
         const { userBookmarksItems, userBookmarksCount, contentTypesMap } = this.state;
         const browsePanelConfig = { id: TAB_BROWSE, panel: FinderPanelComponent, attrs: { sortFieldMappings, sortOrderMappings } };
-        const searchPanelConfig = { id: TAB_SEARCH, panel: SearchPanelComponent };
+        const searchPanelConfig = {
```
Comment (dew326) [inline src/modules/universal-discovery/universal.discovery.module.js]: I think Prettier will make it one line.
Please use Prettier.

**Review by dew326 — CHANGES_REQUESTED** [review]

**Comment by barw4** [comment]
@dew326 resolved in https://github.com/ezsystems/ezplatform-admin-ui-modules/pull/286/commits/6787ebbf3bd09f3b471a2fd7574733021e7978b0

**Review by dew326 — APPROVED** [review]
Could you also create a PR for new UDW in ezplatform-admin-ui?

**Comment by barw4** [comment]
> Could you also create a PR for new UDW in ezplatform-admin-ui?

@dew326 yes, I'll do it after approval of related kernel PR, just to be sure everything is fine.

**Review by GrabowskiM — APPROVED** [review]

**Pull request merged**